From: "Yan, Zheng" <zyan@redhat.com>
Date: Tue, 14 Oct 2014 15:38:01 +0800
Subject: [PATCH] ceph: fix divide-by-zero in __validate_layout()
Git-commit: 0bc62284ee3f2a228c64902ed818b6ba8e04159b
Patch-mainline: v3.18
References: Fate#318586


The 'stripe_unit' field is 64 bits, casting it to 32 bits can result zero.

Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/ioctl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/ioctl.c
+++ b/fs/ceph/ioctl.c
@@ -41,7 +41,7 @@ static long __validate_layout(struct cep
 	/* validate striping parameters */
 	if ((l->object_size & ~PAGE_MASK) ||
 	    (l->stripe_unit & ~PAGE_MASK) ||
-	    (l->stripe_unit != 0 &&
+	    ((unsigned)l->stripe_unit != 0 &&
 	     ((unsigned)l->object_size % (unsigned)l->stripe_unit)))
 		return -EINVAL;
 
